Pirate-Wires/pirate-wires
# X Ads Integration

### `Notion` Ticket
- https://www.notion.so/Twitter-ads-integration-cf4846a0011140529546fc770985a393?pvs=4

### Testing Instructions 
1. Implement the following diff:
```
diff --git a/lib/utils/xAds.ts b/lib/utils/xAds.ts
index 4a1abc9..4937f6a 100644
--- a/lib/utils/xAds.ts
+++ b/lib/utils/xAds.ts
@@ -46,7 +46,7 @@ export const sendXConversionEvent = async ({ eventId, twclid }: { eventId: strin

   const authHeader = oauth.toHeader(oauth.authorize(request_data, token))["Authorization"];

-  await fetch(request_data.url, {
+  const response = await fetch(request_data.url, {
     method: request_data.method,
     headers: {
       "Content-Type": "application/json",
@@ -54,4 +54,8 @@ export const sendXConversionEvent = async ({ eventId, twclid }: { eventId: strin
     },
     body: JSON.stringify(body),
   });
+
+  const json = await response.json();
+
+  console.log(JSON.stringify(json));
 };
```
2. Running the app locally, load any page, and in the url, add the query parameter `twclid=123`
3. Complete the purchase flow (you'll arrive on the `/newsletters` route
4. In the terminal, verify that you see a conversion log such as the one below exactly once:
```
{"request":{"params":{"account_id":"18ce55jlxnt"}},"data":{"conversions_processed":1,"debug_id":"d70e45c2-310b-4f4e-ac1f-4dd234d3ad6a"}}
```
5. Sign out and sign back in
6. Load the `/newsletters` route
7. Verify that you _do not_ see a conversion log in the terminal
8. Sign out
9. Navigate to `/newsletters?twclid=123`
10. Select 1 or more newsletters
11. Enter a valid email in the text bar and click "Sign up"
12. Verify that you see a conversion log in the terminal
